corehq/apps/case_importer/do_import.py

```diff
@@ -0,0 +1,278 @@
+from __future__ import absolute_import, unicode_literals
+
+import uuid
+from collections import namedtuple
+
+from django.utils.functional import cached_property
+
+from casexml.apps.case.const import CASE_TAG_DATE_OPENED
+from casexml.apps.case.mock import CaseBlock, CaseBlockError
+from dimagi.utils.logging import notify_exception
+from soil.progress import set_task_progress
+
+from corehq.apps.export.tasks import add_inferred_export_properties
+from corehq.apps.hqcase.utils import submit_case_blocks
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.users.models import CouchUser
+from corehq.toggles import BULK_UPLOAD_DATE_OPENED
+from corehq.util.datadog.utils import case_load_counter
+from corehq.util.soft_assert import soft_assert
+
+from . import exceptions
+from . import util as importer_util
+from .const import LookupErrors
+
+CASEBLOCK_CHUNKSIZE = 100
+RowAndCase = namedtuple('RowAndCase', ['row', 'case'])
+
+
+class _Importer(object):
+    def __init__(self, domain, config, task, record_form_callback):
+        self.domain = domain
+        self.config = config
+        self.task = task
+        self._record_form_callback = record_form_callback
```

Comment:
I don't follow - do you mean that it'll always be set, and I can just call it directly?  Or that it could be refactored out into the task and completely out of this file?